server/src/main/java/io/deephaven/server/object/ObjectServiceGrpcImpl.java

```diff
@@ -69,15 +141,51 @@ public void fetchObject(
                 .onError(responseObserver)
                 .submit(() -> {
                     final Object o = object.get();
-                    final FetchObjectResponse response = serialize(type, session, o);
-                    responseObserver.onNext(response);
-                    responseObserver.onCompleted();
+                    ObjectType objectTypeInstance = getObjectTypeInstance(type, o);
+
+                    StreamObserver<StreamResponse> wrappedResponseObserver = new StreamObserver<>() {
+                        @Override
+                        public void onNext(StreamResponse value) {
+                            // TODO Do we send a message here, _then_ error if the plugin didn't close? Or do we close
+                            // with an error without sending anything (and release exports)?
```

Comment:
interested in reviewer feedback here - i'm leaning towards send the message _then_ fail the stream to make the exports the user's problem (and at least an old/wrong client can handle the first payload and just not get updates), but i see a strong case for not sending the message at all if the plugin isn't what was expected

Reply:
For case 2 at least, we could (that is, not implemented) fail the _second_ one before it actually exports, and so avoid needing to clean up, but either way, this is "I am writing my code and made a mistake" levels of errors, rather than "this is the sort of thing that will leak during normal usage" sorts of problems. I'd also tend towards "FetchObject is deprecated for removal, within 3 months" since only JS uses it today (and as of this patch no longer calls it at all). Does that change your perspective at all?

Reply:
Good point - I'm not too concerned about edge-case correctness wrt fetch object since it's going away. May be worth removing existing comment and stating as much.

Reply:
I'd say dump the TODO. You're the only implementer, right? I would opt for failing if not already closed. That takes care of both classes of errors, right?